src/lang/lang_types.ml

```diff
@@ -1019,6 +999,11 @@ let rec ( <: ) a b =
     | Ground (Format k), Ground (Format k') -> (
         try Frame_content.merge k k' with _ -> raise (Error (repr a, repr b)) )
     | Ground x, Ground y -> if x <> y then raise (Error (repr a, repr b))
+    | Getter t1, Getter t2 -> (
+        try t1 <: t2 with Error (a, b) -> raise (Error (`Getter a, `Getter b)) )
+    | Arrow ([], t1), Getter t2 -> (
+        try t1 <: t2
+        with Error (a, b) -> raise (Error (`Arrow ([], a), `Getter b)) )
```

Comment:
So, a getter can be unified with a function with no arguments. I see. Are we assuming that, for all practical purposes, they are the same but with a different underlying implementation just like `Fun` and `FFI`?

Reply:
The point it that a function with no argument *is* a valid getter: it is not just the way we implement that, we can actually provide `fun() -> ...` where a getter is expected.